crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -197,22 +197,42 @@ where
         f(state)
     }
 
-    async fn evm_env_at(&self, mut at: BlockId) -> EthResult<(CfgEnv, BlockEnv, BlockId)> {
+    async fn evm_env_at(&self, at: BlockId) -> EthResult<(CfgEnv, BlockEnv, BlockId)> {
         if at.is_pending() {
-            if let Some(pending) = self.client().pending_header()? {
-                let mut cfg = CfgEnv::default();
-                let mut block_env = BlockEnv::default();
-                self.client().fill_block_env_with_header(&mut block_env, &pending.header)?;
-                self.client().fill_cfg_env_with_header(&mut cfg, &pending.header)?;
-                return Ok((cfg, block_env, pending.hash.into()))
-            }
-            // No pending block, use latest
-            at = BlockId::Number(BlockNumberOrTag::Latest);
+            let header = if let Some(pending) = self.client().pending_header()? {
+                pending
+            } else {
+                // no pending block from the CL yet, so we use the latest block and modify the env
+                // values that we can
+                let mut latest = self
+                    .client()
+                    .latest_header()?
+                    .ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+
+                // child block
+                latest.number += 1;
+                // assumed child block is in the next slot
+                latest.timestamp += 12;
```

Comment:
this value is specific to Ethereum's consensus mechanism, would like it extracted to a constant